### ISSUE TITLE
tutorial: fixed path to weight

### ISSUE THREAD

**Issue by shreyas-bk** [zylo117/Yet-Another-EfficientDet-Pytorch#491]
Thanks for your work! I went through the tutorial and had some thoughts

The path to the weight file during evaluation is fixed as follows

`python coco_eval.py -c 0 -p shape -w logs/shape/efficientdet-d0_49_1400.pth`

However, sometimes the file efficientdet-d0_49_1400.pth is not created even after 50 epochs (possibly because the loss increased in the final epoch when I ran it). It would also not be useful if number of epochs is changed.

So wouldn't it be better to have the weight file taken as the latest weight file added to logs/shapes rather than fixing it as efficientdet-d0_49_1400.pth? If so, I can create a pr that does this in 2-3 lines if required. Please let me know!

**Comment by zylo117** [comment]
Thanks for the advice. Actually I think setting a shorter saving interval would do the trick.
https://github.com/zylo117/Yet-Another-EfficientDet-Pytorch/commit/a2aa13e72b37dd3910201f67adc2d908720d296e

**Comment by shreyas-bk** [comment]
Alright thanks. But it still may not solve a potential problem.

Correct me if I'm wrong but because of this code:
`if loss + opt.es_min_delta < best_loss:
                    best_loss = loss
                    best_epoch = epoch

                    save_checkpoint(model, f'efficientdet-d{opt.compound_coef}_{epoch}_{step}.pth')`

There is a chance that 'efficientdet-d0_49_1400.pth' might not be the last weight saved (in case the loss becomes higher than the best_loss), right? In that case, shouldn't the latest saved weight file be used rather than hard-coding the weight file as 'efficientdet-d0_49_1400.pth' Please let me know! If I'm not clear, may I open a pr to show what I mean?

**Comment by zylo117** [comment]
Of course, if you think it's necessary.
I hard coded because I worte it in a hurry and also don't want the coco_eval.py only works on the last weights.

But for now, I think maybe coco_eval.py can load the last saved weights by default and it allows user to specify any weights that need to be tested.

**Comment by shreyas-bk** [comment]
Thanks for your responses and your patience, I appreciate it!

Yes I believe loading the last saved weights would be ideal for the tutorial, as there wouldn't be any errors when it is run, which will be helpful for beginners. I will open a pr shortly for the changes.
